src/fsharp/CompileOps.fs

```diff
@@ -1700,6 +1700,7 @@ let DefaultReferencesForScriptsAndOutOfProjectSources(assumeDotNetFramework) =
           yield "System.Numerics" 
       else
           yield Path.Combine(Path.GetDirectoryName(typeof<System.Object>.Assembly.Location),"mscorlib.dll"); // mscorlib
+          yield typeof<System.Object>.Assembly.Location; // System.Private.CoreLib
```

Comment:
> The type 'Object' is required here and is unavailable. You must add a reference to assembly System.Private.CoreLib

I believe this indicates that the compilation is referencing .NET Core private implementation assemblies, which is now considered invalid

In .NET Framework you used to be able to just reference any assemblies under the framework implementation directory ``c:\windows\...`` . But my understanding is that in the .NET Core world they want all compilations only use facade/reference/... assemblies.  I believe this includes dynamic compilations, which is a bit confusing for me since mentally I always think of .NET Framework dynamic compilations (e.g. ``fsi.exe``) as being against the latest installed .NET as found in the implementation directory.  

I'm not at all clear on where/how .NET Core dynamic code generation for tools like ``fsi.exe`` is expected to find the appropriate reference assemblies, it almost seems to me that they expect a tool like ``fsi.exe`` to download all the nuget packages etc. for a particular framework facade and use those, or find cached ones on the machine.

Because of this, when I last looked at ``fsi.exe`` on .NET Core, I ended up trying to [teach the compiler to respect System.Private.CoreLib as a primary assembly reference for System.Object etc](https://github.com/Microsoft/visualfsharp/pull/1792#discussion_r88955380).  But @KevinRansom told me not to, that this is wrong, and compilation should never reference this assembly.  I'm still confused about what they should reference, and I can really see that in practice people doing dynamic compilation are going to reference the implementation assemblies all the time, just because it's dead simple and you can be guaranteed that they actually exist.  

Another way to put this is that I'm simply not sure what the correct values of ``DefaultReferencesForScriptsAndOutOfProjectSources`` should be for .NET Core, i.e. what the default references for scripts should be.  It feels like they should be the locations of a bunch of .NET Standard 2.0 reference assemblies, but I don't think we can assume they are installed.

Anyway, to help us make progress, let's characterize the specific scenario carefully. I assume it is this:

- You are running the .NET Core version of a tool called Fable.EXE (or whatever its name) 

- Your Fable.EXE uses the .NET Core version of FCS.dll to analyze an F#/Fable script or project, extract its Symbols and Exprs, and generate (print) Javascript code. Your Fable.EXE  doesn't actually do any .NET codegen, but rather JS codegen.  Your Fable.EXE may accept references to .NET and F# assemblies as part of its input specification.

- Your Fable.EXE prefers not to require any additional facade or reference assembly packages.  For the moment it's happy to rely on those returned by evaluating these fragments on .NET Core, taken from ``DefaultReferencesForScriptsAndOutOfProjectSources``:

```
yield Path.Combine(Path.GetDirectoryName(typeof<System.Object>.Assembly.Location),"mscorlib.dll"); // mscorlib
         yield typeof<System.Object>.Assembly.Location; // System.Private.CoreLib"
...
```

- We know these assembly references are "wrong" (they are private implementation assemblies) but they appear to "work" because no actual .NET code generation is done, and the F# compilation logic doesn't really care

- We know that it may be better to reference .NET Standard 2.0 reference assemblies or Fable-specific reference assemblies but that adds complexity to the tool and we don't know where to find these

Is that correct?  BTW have you tried adding the above reference explicitly in ``otherArgs`` of ``GetProjectOptionsFromScript``?

thanks
don